tools/schemacode/pyproject.toml

```
@@ -1,11 +1,64 @@
 [build-system]
-requires = ["setuptools"]
 build-backend = "setuptools.build_meta"
+requires = ["setuptools"]
+
+[project]
+authors = [{name = "bids-standard developers"}]
+classifiers = [
+    "Development Status :: 4 - Beta",
+    "License :: OSI Approved :: MIT License",
+    "Intended Audience :: Science/Research",
+    "Topic :: Scientific/Engineering :: Information Analysis",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: 3.13"
+]
+dependencies = [
+    "click",
+    "pyyaml",
+    "jsonschema"
+]
+description = " Python tools for working with the BIDS schema."
+dynamic = ["version"]
+license = {text = "MIT"}
+maintainers = [{name = "bids-standard developers", email = " bids.maintenance@gmail.com"}]
+name = "bidsschematools"
+readme = "README.md"
+requires-python = ">=3.9"
+
+[project.optional-dependencies]
+all = [
+    "bidsschematools[doc,render,tests,expressions]"
+]
+doc = [
+    "sphinx>=1.5.3",
+    "sphinx_rtd_theme"
+]
+expressions = ["pyparsing"]
+render = [
+    "tabulate",
+    "pandas",
+    "markdown-it-py"
+]
+tests = [
+    "codecov",
+    "coverage[toml]",
+    "flake8",
+    "flake8-black",
+    "flake8-isort",
+    "pytest>6",
+    "pytest-cov"
+]
```

Ordering: Optional functionality, tests, docs, all. Also, tests will fail if we don't include optional functionality, so adding that.

```suggestion
expressions = ["pyparsing"]
render = [
    "tabulate",
    "pandas",
    "markdown-it-py"
]
tests = [
    "bidsschematools[expressions,render]",
    "codecov",
    "coverage[toml]",
    "flake8",
    "flake8-black",
    "flake8-isort",
    "pytest>6",
    "pytest-cov"
]
doc = [
    "sphinx>=1.5.3",
    "sphinx_rtd_theme"
]
all = [
    "bidsschematools[doc,tests]"
]
```